sdk/monitor/Azure.Monitor.OpenTelemetry.Exporter/src/Customizations/Models/TelemetryItem.cs

```diff
@@ -25,18 +25,21 @@ internal partial class TelemetryItem
 
             if (activity.GetTelemetryType() == TelemetryType.Request)
             {
-                if (activityTagsProcessor.activityType.HasFlag(OperationType.V2))
+                var displayName = activity.DisplayName;
+                if (!string.IsNullOrEmpty(displayName)
+                    && !displayName.StartsWith(activity.Source.Name))
```

Comment:
This check is intended to handle detecting whether the DisplayName has been set to a value different than the value it started with. We know from the source code that the ASP.NET core request activity starts with the name `Microsoft.AspNetCore.Hosting.HttpRequestIn` and comes from activity source `Microsoft.AspNetCore`, and there's a bunch of other code that depends on that.

Reply:
> We know from the source code that the ASP.NET core request activity starts with the name Microsoft.AspNetCore.Hosting.HttpRequestIn

This may be tricky. Asp.Net Core is **not the only** instrumentation possible, and *this* exporter shouldn't bake in any assumptions about instrumentations.

Reply:
I agree with you @cijothomas . I think the "right" fix for this is just to use the `activity.DisplayName`, and skip all the other conditions. As I tried to explain, they are unnecessary b/c the same values are set by the AspNetCore instrumentation. But if I'd made that fix I would have broken a number of tests that enforce that (redundant) logic.

Also, I was trying to come up with a way to determine whether _any other_ instrumentation library has set the value. This does work for that problem. My thinking is: If the developer or any other instrumentation library has set the `DisplayName` value, it should not be overwritten by the AspNetCore instrumentation (so this is a possible way to detect changes to fix https://github.com/open-telemetry/opentelemetry-dotnet-contrib/issues/1948 ).